influence_benchmark/stats/preferences_per_iteration.py

```diff
@@ -33,37 +33,64 @@ def compute_average_traj_rewards(traj_timestep_df):
     return avg_rewards_df
 
 
-def get_best_worst_n_trajectories(traj_path: Path, num_chosen_trajs: int) -> Tuple[List[Dict], List[Dict]]:
-    top_n_dict = get_func_n_trajectories(traj_path, num_chosen_trajs, pd.DataFrame.nlargest)
-    bottom_n_dict = get_func_n_trajectories(traj_path, num_chosen_trajs, pd.DataFrame.nsmallest)
+def compute_final_turn_rewards(traj_timestep_df):
+    # Get the final turn for each trajectory
+    final_turn_df = (
+        traj_timestep_df.groupby(["env_name", "initial_state_id", "trajectory_id"])
+        .apply(lambda x: x.loc[x["turn"].idxmax()])
+        .reset_index(drop=True)
+    )
+
+    # Select the reward and influence level from the final turn
+    final_rewards_df = final_turn_df[
+        ["env_name", "initial_state_id", "trajectory_id", "timestep_reward", "timestep_influence_level"]
+    ].rename(columns={"timestep_reward": "traj_final_rew", "timestep_influence_level": "traj_final_infl"})
+
+    return final_rewards_df
```

Comment:
Just to double check, when using the finel_reward, we are also changing the preference model prompt to evaluate the whole conversation, right?

Reply:
Well sometimes haha, we probably should try and make sure they match yes